Error when no binary_sensor configured

When creating a device configuration where no binary_sensor is configured an error is thrown during compilation:

```
Compiling .pioenvs\ithodaalderoptester\src\esphome\components\modbus_spy\modbus_frame.cpp.o
Compiling .pioenvs\ithodaalderoptester\src\esphome\components\modbus_spy\modbus_frame_detector_factory.cpp.o
Compiling .pioenvs\ithodaalderoptester\src\esphome\components\modbus_spy\modbus_register_sensor.cpp.o
src/esphome/components/modbus_spy/modbus_binary_sensor.cpp:6:10: fatal error: esphome/components/binary_sensor/binary_sensor.h: No such file or directory
 #include "esphome/components/binary_sensor/binary_sensor.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
Compiling .pioenvs\ithodaalderoptester\src\esphome\components\modbus_spy\modbus_request_detector.cpp.o
*** [.pioenvs\ithodaalderoptester\src\esphome\components\modbus_spy\modbus_binary_sensor.cpp.o] Error 1
In file included from src/esphome/components/modbus_spy/modbus_data_publisher.h:13,
                 from src/esphome/components/modbus_spy/modbus_data_publisher.cpp:5:
src/esphome/components/modbus_spy/modbus_binary_sensor.h:7:10: fatal error: esphome/components/binary_sensor/binary_sensor.h: No such file or directory
 #include "esphome/components/binary_sensor/binary_sensor.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
*** [.pioenvs\ithodaalderoptester\src\esphome\components\modbus_spy\modbus_data_publisher.cpp.o] Error 1
```

Example configuration which will result in error:
```
esphome:
  name: daalderop
  friendly_name: Daalderop

esp32:
  board: esp32dev
  framework:
    type: arduino

external_components:
  - source: github://pdjong/modbus-spy-esphome@latest-stable

uart:
  baud_rate: 19200
  rx_pin: GPIO32
  parity: ODD

modbus_spy:
  id: modbusspy

sensor:
  - platform: modbus_spy
    name: "Inside Temperature"
    device_address: 0x0B
    register_address: 41200
```

--- a/components/modbus_spy/__init__.py
+++ b/components/modbus_spy/__init__.py
@@ -6,6 +6,7 @@ from esphome.components import uart
 CODEOWNERS = ["@pdjong"]
 
 DEPENDENCIES = ["uart"]
+AUTO_LOAD = ["binary_sensor"]
 MULTI_CONF = True
 
 modbus_spy_ns = cg.esphome_ns.namespace('modbus_spy')

